### PR TITLE
Task 2: [Standard Libraries] Created mainOSProcess script, adding git…

### PR DESCRIPTION
**Pull request by mykolaantypenko1999**
Task 2: [Standard Libraries] Created mainOSProcess script, adding gitignore file, updating package.json

### CODE REVIEW

File: mainOSProcess.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    updateRate = 1000;
+}
+
+console.log(`Info about OS \nPlatform: ${os.platform()} \nRAM: ${os.totalmem()} \nCPU architecture: ${os.arch()}`);
+console.log('\n\x1b[94mThe most CPU-intensive process in OS');
+console.log('\x1b[37m');
+
+// Function of writing OS process
+function writeProcessDetails() {
+    exec(command, (error, stdout, stderr) => {
```
Comment (BorysLypivets) [inline mainOSProcess.js:29]: [Optional task]
please use the function `promisify()` of the standard node.js module `util` to rewrite your program in async/await syntax. 

P.S. You are welcome you choose what to use for rewriting for `fs.appendFile()`: `util.promisify()` or `fs.promises.appendFile()`. 
But the second variant will bring you more respect and glory 😉

**Review by BorysLypivets — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "description": "NodeJS Mentoring Program App",
   "main": "index.js",
   "scripts": {
-    "start": "nodemon index.js"
+    "start": "nodemon index.js",
```
[the remark this answers is not in the document]
Comment (mykolaantypenko1999) [reply package.json]: Fixed

File: package.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
   "description": "NodeJS Mentoring Program App",
   "main": "index.js",
```
[the remark this answers is not in the document]
Comment (mykolaantypenko1999) [reply package.json]: Fixed

File: mainOSProcess.js (hunk 1 above)
Comment (mykolaantypenko1999) [reply mainOSProcess.js]: An additional task was completed

**Review by BorysLypivets — CHANGES_REQUESTED** [review]

File: mainOSProcess.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+} else if (os.platform() === windowsOS) {
+    command = 'powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + \' \' + $_.CPU + \' \' + $_.WorkingSet }"';
+    updateRate = 1000;
+}
+
+console.log(`Info about OS \nPlatform: ${os.platform()} \nRAM: ${os.totalmem()} \nCPU architecture: ${os.arch()}`);
+console.log('\n\x1b[94mThe most CPU-intensive process in OS');
+console.log('\x1b[37m');
+
+const execAsync = util.promisify(exec);
```
[the remark this answers is not in the document]
Comment (mykolaantypenko1999) [reply mainOSProcess.js]: Fixed

File: mainOSProcess.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    // Current time in unixtime format
+    const currentTimeInUnixTime = Math.floor(new Date().getTime() / 1000);
+    // Format for the output information
+    const logInput = `${currentTimeInUnixTime}: ${stdout.trim()}`;
+    // Write information to console in one line
+    process.stdout.write(`\r${logInput}`);
+
+    // One per minute write process to log file
+    if (currentTimeInUnixTime % 60 === 0) {
+        fs.promises.appendFile(logFileName, `${logInput}\n`)
```
[the remark this answers is not in the document]
Comment (mykolaantypenko1999) [reply mainOSProcess.js]: Refactored

**Review by BorysLypivets — APPROVED** [review]

**Pull request merged**